tests/Unit/Execution/Arguments/ArgumentSetTest.php

```diff
@@ -120,4 +126,64 @@ public function testListOfInputObjectsToArray(): void
             $argumentSet->toArray()
         );
     }
+
+    public function testRenameInput(): void
+    {
+        $generateRenameDirective = function ($newName) {
```

Comment:
Using the `PartialParser` is much more simple and terse then this.

Reply:
@spawnia I think I need some direction on this. I've looked into the `PartialParser` and have attempted to use it. What's confusing is that when I use `PartialParser::directive` I get a `DirectiveNode`, as opposed to a `BaseDirective` descendent. In fact, I can't find the `RenameDirective` anywhere in the result when using that. I also tried using `PartialParser::InputObjectTypeDefinition`, but this also returns nothing but nodes.

The `@spread` test isn't using the `PartialParser`, but is just instantiating the directive, which is likely fine because it doesn't need any arguments. I clearly need to access the `RenameDirective` within the `ArgumentSet::rename()` method, but when I parse inputs with the directive I only get nodes.

I'm clearly missing something, here. Can you please help me to understand where nodes connect with the Lighthouse `BaseDirective` descendents?

Reply:
Hey @JasonTheAdams I'd like to help you to also continue learning.. may I touch your PR?

Reply:
@enzonotario I don't mind at all! I look forward to seeing what you change. If you wouldn't mind, please provide some explanation so I can learn for the future. 😄

Reply:
well I don't promise anything but will try todo my best! :)

Reply:
Hey folks, I have changed this to using PartialParser, although for now I don't really understand at all about the directive generation, but it works. I hope it is ok. Otherwise, just sorry :)

Reply:
The `BaseDirective` requires a parsed AST definition of a directive under the hood to read the arguments from. The easiest way is to `new` up an instance of whatever directive you want to test and `hydrate` it with the result of a call to `PartialParser`.